### DISCUSSION TITLE
Multiple project support

### DISCUSSION THREAD

**Discussion by indykoning** [buggregator/server#148] [Ideas]
When developing multiple applications for a single project this works fantastic, not only aggregating the different outputs from debugging software but also aggregating of these for the applications so you have a good timeline of what each application was up to when debugging.

Running this locally you have no choice but aggregate every project into the same overview, unless you run multiple buggregator instances on different ports.

I think it would be nice to have a dropdown to switch projects in buggregator, kind of like how you can have different projects in Sentry.
Some ways we could separate events into projects are:
- auth username and/or password (Ray, SMTP, Profiler. Inspector, Sentry, Http Dumps)
- url variable (Inspector, Sentry)
- ? (monolog, VarDumper)

e.g. Profiler also supports an application name & Inspector supports an ingestion key

**Comment by roxblnfk** [comment]
> ? (monolog, VarDumper)

For vardumps, I can add a hostname or any other context in case of using `buggregator/trap`

**Comment by butschster** [comment]
Yeah, we used to have project support, but then I did a big refactor and had to disable it. Check out [HandleReceivedEvent](https://github.com/buggregator/server/blob/master/app/src/Application/Commands/HandleReceivedEvent.php#L18) in that command – you'll see there's a project ID property.

Right now, we're putting together guidelines on how to contribute and get involved with Buggregator development. If you've got some free time, we'd love to have you on board. It'd be awesome if you could create an issue at least, and I'll make sure it gets added to our roadmap.

The frontend development is a bit tricky because it's not syncing up with the server side as smoothly as we'd like. Managing the project list from the frontend is a bit of a headache. I'm thinking of adding support to handle it using `yaml` files and then mounting them inside the Docker container.

**Reply by indykoning** [reply]
Awesome! good to see that the groundwork is still there!

I can't make any promises because I'm not familiar in Spiral  but if I can find some free time I will try and take a look for this tool that has become invaluable to me!

I have started talk at work to bring some compatibility in Magento 2 as well, which we've already made a Sentry module https://github.com/justbetter/magento2-sentry

**Reply by butschster** [reply]
Thanks a bunch for your willingness to help out! I've decided to take on the task of implementing Projects myself, but your offer means a lot.

By the way, it would be awesome if you could contribute some information about integrating Buggregator with your package. Adding details about your package for Sentry integration with Magento to our docs would be cool. If you're up for it.

**Comment by butschster** [comment]
Hi @indykoning!

I've just released a beta release of Buggregator: [link to GitHub release](https://github.com/buggregator/server/pkgs/container/server/209719651?tag=1.8.0-beta1). You can install it with the `dev` Docker image tag. In this release, I've added projects on the backend side, but the frontend does not support them yet.
